### ISSUE TITLE
LeafNotFound when trying API call mmr_generateProof

### ISSUE THREAD

**Issue by forgetso** [paritytech/substrate#13459]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

Running substrate `polkadot-v0.9.38`, no mmr proofs are returned from this API call.

```
(base) ➜  substrate git:(polkadot-v0.9.38) ✗ ./target/release/substrate  --dev -d ./chain-data --unsafe-ws-external --rpc-external --prometheus-external
2023-02-24 10:21:54 Substrate Node    
2023-02-24 10:21:54 ✌️  version 3.0.0-dev-18bb7c7c841    
2023-02-24 10:21:54 ❤️  by Parity Technologies <admin@parity.io>, 2017-2023    
2023-02-24 10:21:54 📋 Chain specification: Development    
2023-02-24 10:21:54 🏷  Node name: boorish-queen-2636    
2023-02-24 10:21:54 👤 Role: AUTHORITY    
2023-02-24 10:21:54 💾 Database: RocksDb at ./chain-data/chains/dev/db/full    
2023-02-24 10:21:54 ⛓  Native runtime: node-268 (substrate-node-0.tx2.au10)    
2023-02-24 10:21:57 [0] 💸 generated 1 npos voters, 1 from validators and 0 nominators    
2023-02-24 10:21:57 [0] 💸 generated 1 npos targets    
2023-02-24 10:21:59 Using default protocol ID "sup" because none is configured in the chain specs    
2023-02-24 10:21:59 🏷  Local node identity is: 12D3KooWKGrthLCDH3rK3R9W7BfMLbrET6eaVAFsw5ES9jEAsnzn    
...
2023-02-24 10:21:59 📦 Highest known block at #11466    
...
2023-02-24 10:21:59 ⚠️  The hardware does not meet the minimal requirements for role 'Authority'.    
2023-02-24 10:21:59 👶 Starting BABE Authorship worker    
2023-02-24 10:22:00 🙌 Starting consensus session on top of parent 0x2ce8470ddb3da7489cda06353a082b3b64f951dab1fb89b85e09afd27b33d56a    
2023-02-24 10:22:00 🎁 Prepared block for proposing at 11467 (1 ms) [hash: 0xe7574967a662fc7be930aadb5a25fe3d18e2ef57b3f72d7801487696706a6f87; parent_hash: 0x2ce8…d56a; extrinsics (1): [0x88a6…45d1]]    
...
```

```
block=11271; bestblock=11271; statequeryblock=11272; statequeryblockhash=`curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "chain_getBlockHash", "params":['$statequeryblock']}' localhost:9933 | jq '.result'`; curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "mmr_generateProof", "params":[['$block'], '$bestblock', '$statequeryblockhash']}' localhost:9933
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   177  100   102  100    75    99k  75000 --:--:-- --:--:-- --:--:--  172k
{"jsonrpc":"2.0","error":{"code":8011,"message":"Leaf was not found","data":"Error::LeafNotFound"},"id":1}%  
```

```
block=20; bestblock=20; statequeryblock=11400; statequeryblockhash=`curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "chain_getBlockHash", "params":['$statequeryblock']}' localhost:9933 | jq '.result'`; curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "mmr_generateProof", "params":[['$block'], '$bestblock', '$statequeryblockhash']}' localhost:9933
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   177  100   102  100    75    99k  75000 --:--:-- --:--:-- --:--:--  172k
{"jsonrpc":"2.0","error":{"code":8011,"message":"Leaf was not found","data":"Error::LeafNotFound"},"id":1}%   
```

Does this API call work with the dev chain?

### Steps to reproduce

- Start substrate in `--dev` mode
- Run API calls shown above

**Comment by bkchr** [comment]
CC @acatangiu

**Comment by acatangiu** [comment]
I believe this is duplicate of https://github.com/paritytech/polkadot/issues/6648#issuecomment-1410090352. Please reopen if that's not the case.

TLDR: there was a breaking change to RPC API calling into Runtime API that we decided not to version as previous API was some beta version that should not be used going forward.

It should start working again with nodes >= `polkadot-v0.9.38` and runtimes >= `9380`

**Comment by forgetso** [comment]
Thanks @acatangiu it's working for me with `polkadot-v0.9.39`
